src/nvim/lua/executor.c

```diff
@@ -1605,9 +1611,10 @@ Object nlua_call_ref(LuaRef ref, const char *name, Array args, LuaRetMode mode,
     nlua_push_Object(lstate, &args.items[i], 0);
   }
 
-  if (nlua_pcall(lstate, nargs, 1)) {
-    // if err is passed, the caller will deal with the error.
-    if (err) {
+  bool fast = name && strcmp(name, "msg_show") == 0;
+  if (fast ? nlua_fast_cfpcall(lstate, nargs, 1, -1) : nlua_pcall(lstate, nargs, 1)) {
```

Comment:
It probably will look better to structure this like `if (fast) { ... the fast stuff } else { the old code}`